src/install/fetch.ts

```diff
@@ -167,13 +167,25 @@ type FetchToStoreOptions = {
  */
 function fetchToStoreCached (opts: FetchToStoreOptions): Promise<void> {
   return memoize(opts.fetchLocks, opts.resolution.id, async function () {
-    if (opts.force || !await exists(opts.target)) {
-      await rimraf(opts.target)
+    const target = opts.target
+    const targetStage = `${opts.target}-stage`
```

Comment:
This is a valid version name. There might be an error when installing 1.0.0. Later someone might try to install 1.0.0-stage. The broken 1.0.0 fetch will be considered to be 1.0.0-stage